src/mono/mono/mini/mini-generic-sharing.c

```diff
@@ -1246,6 +1246,16 @@ get_wrapper_shared_vtype (MonoType *t)
 	}
 	g_assert (tuple_class);
 
+	for (int i = 0; i < findex; i++) {
+		if (m_type_is_byref (args [i])) {
+#if TARGET_SIZEOF_VOID_P == 8
+			args [i] = m_class_get_byval_arg (mono_defaults.int_class);
+#else
```

Comment:
Couldn't this be int_class in both cases ?

Reply:
Or int32/int64 for more sharing ?

Reply:
Was following the logic from https://github.com/dotnet/runtime/blob/main/src/mono/mono/mini/mini-generic-sharing.c#L1296 so I don't happen to use different types and prevent sharing